### PR TITLE
8344041: Re-enable external specs page

### PR DESCRIPTION
**Pull request by hns**
Please review a change to enable the External Specifications page in API docs. The page was disabled because of missing `@spec` tags in OpenJDK sources. This is a straightforward undo of of #13127 / [JDK-8304689], which means the hidden ad-hoc `--no-external-specs-page` option is also removed.

[JDK-8304689]: https://bugs.openjdk.org/browse/JDK-8304689

[This is what the External Specifications page looks like.](https://cr.openjdk.org/~hannesw/8344041/api.00/external-specs.html)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8344041](https://bugs.openjdk.org/browse/JDK-8344041): Re-enable external specs page (**Enhancement** - P3)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Nizar Benalla](https://openjdk.org/census#nbenalla) (@nizarbenalla - Committer)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22240/head:pull/22240` \
`$ git checkout pull/22240`

Update a local copy of the PR: \
`$ git checkout pull/22240` \
`$ git pull https://git.openjdk.org/jdk.git pull/22240/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22240`

View PR using the GUI difftool: \
`$ git pr show -t 22240`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22240.diff">https://git.openjdk.org/jdk/pull/22240.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22240#issuecomment-2486089963)
</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Build change looks good.

**Comment by nizarbenalla** [comment]
This looks like a straight forward undo of https://github.com/openjdk/jdk/commit/2f34687ec1210317a9215496d0184fe30bda6437, while keeping relevant doc changes.

**Review by nizarbenalla — APPROVED** [review]

File: test/langtools/jdk/javadoc/doclet/testSpecTag/TestSpecTag.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 /*
  * @test
- * @bug 6251738 8226279 8297802 8296546 8305407
+ * @bug 6251738 8226279 8297802 8305407
```
Comment (nizarbenalla) [inline test/langtools/jdk/javadoc/doclet/testSpecTag/TestSpecTag.java:26]: Nice catch! It seems this bug id was added by mistake, it does not match the id of the JBS issue where it was added.

**Review by liach — APPROVED** [review]

**Pull request closed without merging**